docs: Remove outdated apmpackage doc mention (#442)
elastic/apm-data: dev_docs/HOW_TO.md

         go mod edit -replace=github.com/elastic/apm-data=/path/to/your/apm-data
         make update

> 2. Modify the [`apm-data` plugin](https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/apm-data) to add the field to Elasticsearch mapping. Note that since the default mappings is [dynamic](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/apm-data/src/main/resources/component-templates/apm%40mappings.yaml#L8), you may not even need to update the plugin if the field can be mapped dynamically.

Reviewer: ```suggestion
### 3. Make changes in Elasticsearch [apm-data plugin](https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/apm-data)
1. Modify the [`apm-data` plugin](https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/apm-data) to add the field to Elasticsearch mapping. Note that since the default mappings is [dynamic](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/apm-data/src/main/resources/component-templates/apm%40mappings.yaml#L8), you may not even need to update the plugin if the field can be mapped dynamically.
```

I'm looking at the changes with fresh eyes. We'll need to make it very clear the distinction between apm-data repo and apm-data plugin in ES. The header `## 2. Make changes in [apm-server]` is incorrect here. It should be something like `Make changes in Elasticsearch apm-data plugin`.

   1. Find the corresponding data stream directory in `src/main/resources/component-templates/`. If the field applies to multiple data streams (e.g. `processor.event`), make sure all the corresponding data streams are updated. 
   2. Add the field to the YAML file for the data stream, e.g. `src/main/resources/component-templates/traces-apm@mappings.yaml`.
   3. In case any changes to ingest pipelines and ILM policies are needed, they are inside `src/main/resources/ingest-pipelines/` and `src/main/resources/lifecycle-policies/` respectively.

>    4. Finally, you will need to build ES docker image in local ES repo with `./gradlew buildDockerImage` and reference it in apm-server `docker-compose.yml`.

Reviewer: like the "apm-data go mod replace", these steps are to facilitate local testing only. The apm-server PR shouldn't contain these bumps, especially the output of `./gradlew buildDockerImage` because the image is not published. Can you clarify the steps, which of them are for local testing, and what should be included in apm-server PR?

         go mod edit -replace=github.com/elastic/apm-data=/path/to/your/apm-data
         make update

> 2. Update the [`apm-data` plugin](https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/apm-data) accordingly.

Author: For reviewers: Need someone with experience on this to add further explanations if possible.

Reviewer: Also another difference between apmpackage in apm-server and apm-data plugin in ES is that apm-data plugin uses `dynamic: true` (see `apm@mappings.yaml`). If a proposed new field can be mapped dynamically, we may not need to change apm-data plugin at all.

> ### 3. Test your changes with system test (in apm-server)

Reviewer: Since apm-data plugin changes won't show up in system test, we'll need to build ES docker image locally `./gradlew buildDockerImage` in ES repo, and reference it in apm-server docker compose yml before system test starts.

In section 5 Finalize PRs, mention that ES apm-data plugin PR needs to be reviewed and merged, apm-server needs to get a ES docker image bump and apm-data bump, then only after that the new system tests in apm-server will pass, and apm-server PR can be merged.

### 5. Finalize PRs

> 1. If you have modified the `apm-data` plugin on ES, have that PR reviewed and merged first.

Reviewer: ```suggestion
1. If you have modified the `apm-data` plugin in ES, have that PR reviewed and merged first. Wait for a new ES build, and wait for [apm-server automation](https://github.com/elastic/apm-server/blob/main/.github/workflows/bump-elastic-stack.yml) to bump stack version.
```

> 2. Create a PR in apm-data, and have it reviewed and merged.

Reviewer: ```suggestion
2. Create a PR in apm-data, and have it reviewed and merged. Follow the instructions in the repo to release a new version, otherwise apm-server automation to bump apm-data version won't work.
```

> 3. In apm-server, bump apm-data dependency (plus ES image if applicable) and create a PR. Note that usually, the bots will bump them automatically, so there's no need for you to do it, but this is included here for reference.

Reviewer: ```suggestion
3. In apm-server, bump apm-data dependency and create a PR. Note that usually, the bots will bump them automatically, so there's no need for you to do it, but this is included here for reference.
```

### Example set of PRs:
- [ES plugin PR](https://github.com/elastic/elasticsearch/pull/112440)
- [apm-data PR](https://github.com/elastic/apm-data/pull/362)

> - apm-server did not have PR since it's bumped by bots 

Reviewer: ```suggestion
- apm-server dependency bump PRs from automation:
   - [apm-data bump](https://github.com/elastic/apm-server/pull/15702)
   - [ES image bump](https://github.com/elastic/apm-server/pull/15813)
```
